lib/edgehog_device_forwarder/connections.ex

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.Connections do
+  use Supervisor
+
+  alias EdgehogDeviceForwarder.Connections.TokenData
+
+  @tokens Application.compile_env!(:edgehog_device_forwarder, EdgehogDeviceForwarder.Connections)[
+            :tokens
+          ]
+
+  @type token :: String.t()
+  @type device_id :: TokenData.device_id()
+  @type realm :: TokenData.realm()
+  @type socket :: TokenData.socket()
+
+  # API
+
+  ## Token
+  @spec register_device_info(token, %{:device_id => device_id, :realm => realm}) ::
+          :ok | {:error, :token_not_found}
+  def register_device_info(token, device_info) do
```

Comment:
Since `device_info` it's just 2 things I'm not sure passing it as a free form map is the best thing to do here. I'd expect you to either unpack this into 2 different `device_id` and `realm` arguments or at least use a struct. This applies all the way down to `put_device_info` too.